### PR TITLE
REFACTOR: Update schema for service

### PR DESCRIPTION
**Pull request by bennettbuchanan**
Adds the preliminary changes to the schema to allow for a configurable service level. `service` is set using a property `component` passed to the Utapi configuration. See https://github.com/scality/S3/pull/537.

### CODE REVIEW

**Comment by bennettbuchanan** [comment]
@ironman-machine

**Comment by bennettbuchanan** [comment]
e2e: http://ci.ironmann.io/gh/scality/Integration/9152

**Comment by bennettbuchanan** [comment]
http://ci.ironmann.io/gh/scality/Integration/9477

**Comment by bennettbuchanan** [comment]
http://ci.ironmann.io/gh/scality/Integration/9486

**Comment by bennettbuchanan** [comment]
passed on rebuild: http://ci.ironmann.io/gh/scality/Integration/9491

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

File: src/lib/ListMetrics.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ export default class ListMetrics {
     /**
      * Assign the metric property to an instance of this class
      * @param {string} metric - The metric type (e.g., 'buckets', 'accounts')
+     * @param {string} component - The service component (e.g., 's3')
      */
     constructor(metric) {
```
Comment (ploki) [inline src/lib/ListMetrics.js]: no ```component``` parameter here

File: src/lib/ListMetrics.js [hunk 2]
```diff
@@ -19,26 +20,29 @@ export default class ListMetrics {
...
         const obj = {};
-        obj[type] = resource;
+        const schemaKeys = {
+            buckets: 'bucket',
+            accounts: 'accountId',
+        };
+        obj[schemaKeys[this.metric]] = resource;
         obj.level = this.metric;
+        // Include service to generate key for metric
+        obj.service = component;
```
Comment (ploki) [inline src/lib/ListMetrics.js]: component or service?

Comment (bennettbuchanan) [inline src/lib/ListMetrics.js]: In the S3 config, this property is called `component`, I suppose internally to Utapi it should be just be `service`.

File: src/lib/ListMetrics.js (hunk 1 above)
Comment (bennettbuchanan) [inline src/lib/ListMetrics.js]: Good catch. Thanks!

**Comment by bennettbuchanan** [comment]
Closing this PR and including changes made here in PR https://github.com/scality/utapi/pull/117 which moreover introduces complete support for service-level metrics. This allows for Integration testing of service-level metrics.

**Pull request closed without merging**